api/src/main/resources/plugins/flink.py

```diff
@@ -0,0 +1,176 @@
+"""
+Name:       flink.py
+Purpose:    Submits flink batch/streaming jobs based on application package
+Author:     PNDA team
+
+Created:    02/03/2018
+
+Copyright (c) 2016 Cisco and/or its affiliates.
+
+This software is licensed to you under the terms of the Apache License, Version 2.0 (the "License").
+You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+The code, technical concepts, and all information contained herein, are the property of Cisco Technology, Inc.
+and/or its affiliated entities, under various laws including copyright, international treaties, patent,
+and/or contract. Any use of the material herein must be in accordance with the terms of the License.
+All rights not expressly granted by the License are reserved.
+
+Unless required by applicable law or agreed to separately in writing, software distributed under the
+License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+either express or implied.
+"""
+
+# pylint: disable=C0103
+
+import json
+import os
+import logging
+import platform
+from shutil import copy
+import deployer_utils
+from plugins.base_creator import Creator
+
+
+class FlinkCreator(Creator):
+
+    def validate_component(self, component):
+        errors = []
+        file_list = component['component_detail']
+        if 'application.properties' not in file_list:
+            errors.append('missing file application.properties')
+        if 'upstart.conf' in file_list:
+            errors.append('Support for user supplied upstart.conf files has been deprecated, ' +
```

Comment:
Flink has never supported this, so no need to check for packages like this.

Reply:
Removed condition to check the upstart.conf.